### PR TITLE
Deployed gateway tests

### PR DESCRIPTION
**Pull request by Shillaker**

### CODE REVIEW

File: .github/workflows/test-integration.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setup-cli
+        with:
+          access-key: ${{ secrets.SCW_ACCESS_KEY }}
+          secret-key: ${{ secrets.SCW_SECRET_KEY }}
+          project-id: ${{ secrets.SCW_DEFAULT_PROJECT_ID }}
+          organization-id: ${{ secrets.SCW_DEFAULT_ORGANIZATION_ID }}
```
Comment (Shillaker) [inline .github/workflows/test-integration.yml:19]: It's a shame we have to repeat this, but it's worth it to run the setups in parallel. Can't put this in the action as actions don't have access to secrets.

File: cli/tests/integration/environment.py [hunk 2]
```diff
@@ -30,9 +34,18 @@ def get_docker_compose_env():
 
     @staticmethod
     def get_scw_env():
-        func_a_url = f'https://{os.environ["FUNC_A_DOMAIN"]}:443'
-        gw_admin_url = f'https://{os.environ["GATEWAY_ADMIN_HOST"]}:443'
-        gw_url = f'https://{os.environ["GATEWAY_HOST"]}:443'
```
Comment (Shillaker) [inline cli/tests/integration/environment.py:35]: This used to get set by the `Makefile` but now we can look it up dynamically

File: cli/cli/infra.py [hunk 3]
```diff
@@ -64,10 +65,11 @@
 class InfraManager(object):
     def __init__(self):
         # Initialise SCW client
-        self.scw_client = Client.from_config_file_and_env(profile_name="dev")
+        self.scw_client = Client.from_config_file_and_env()
```
Comment (cyclimse) [inline cli/cli/infra.py:68]: We could also add a CLI flag for this

File: cli/cli/infra.py [hunk 4]
```diff
@@ -159,6 +161,26 @@ def _get_namespace(self):
...
+        namespace = namespaces.namespaces[0]
+        functions: ListFunctionsResponse = self.functions.list_functions(
+            namespace_id=namespace.id, region=API_REGION
+        )
+
+        for f in functions.functions:
+            if f.name == function_name:
+                return f.domain_name
+
+        return None
```
Comment (cyclimse) [inline cli/cli/infra.py:182]: (If you use the name filter from the API:)

```suggestion
   return functions[0] if functions else None
```

**Review by cyclimse — APPROVED** [review]

File: cli/cli/infra.py (hunk 3 above)
Comment (Shillaker) [reply cli/cli/infra.py:68]: Yes that would be nice, would need to add it to all the commands that use the API which would be a lot of changes so won't include in this PR.

File: cli/cli/infra.py [hunk 5]
```diff
@@ -159,6 +161,26 @@ def _get_namespace(self):
...
+            region=API_REGION,
+            name=namespace_name,
+        )
+        if len(namespaces.namespaces) == 0:
+            click.secho("No namespace found", fg="red", bold=True)
+            raise click.Abort()
+
+        namespace = namespaces.namespaces[0]
+        functions: ListFunctionsResponse = self.functions.list_functions(
+            namespace_id=namespace.id, region=API_REGION
```
[the remark this answers is not in the document]
Comment (Shillaker) [reply cli/cli/infra.py:175]: Yes good point, I actually used that for the function lookup so I can refactor all of these.

**Pull request merged**